### PR TITLE
fix: improve generateStackTrace function to hide vue warn message

### LINKED ISSUE #996: [Vue warn]: Property ("_isVue" | "constructor") was accessed during render but is not defined on instance.
I'm using honeybadger with Laravel, Inertia and Vue3.

Honeybadger seems to be accessing a Vue2 deprecated property.

I found the equivalent issue in Sentry while searching the error
https://github.com/getsentry/sentry-javascript/issues/4170

And the PR that fixed it:
https://github.com/getsentry/sentry-javascript/pull/4461/files

## What are the steps to reproduce this issue?
1. The error happened when calling `includes` on `null` in a computed property.

## What happens?
Some errors but not others make it so honeybadger throws its own error. 
`[Vue warn]: Property "_isVue" was accessed during render but is not defined on instance.`

## What were you expecting to happen?
…

## Any logs, error output, etc?
Following the stacktrace in the browser points to this line:
`Function.prototype.apply.call(original, _window.console, arguments);`

## Any other comments?

Here are the places where it's used:

https://github.com/honeybadger-io/honeybadger-js/blob/317220310f07edd18283214fe8901cab31218416/packages/vue/src/vue-debug.js#L17

https://github.com/honeybadger-io/honeybadger-js/blob/317220310f07edd18283214fe8901cab31218416/packages/vue/src/vue-debug.js#L44

## What versions are you using?
**Operating System:** Windows 10
**Package Name:**  @honeybadger-io/vue  
**Package Version:** 4.8.2
**Browser Version:** Chrome

### PR DESCRIPTION
**Pull request by subzero10**
## Status
**READY**
Fixes: #996

## Description
Improves `generateStackTrace` function to hide `Vue[warn]` message about undefined property `_isVue`.

*Bonus*: adds `jest-fetch-mock` to remove some errors logged in tests

### CODE REVIEW

File: packages/vue/src/vue-debug.js [hunk 1]
```diff
@@ -8,15 +8,27 @@ const classify = str => str
...
   if (vm.$root === vm) {
-    return '<Root>'
+    return ROOT_COMPONENT
+  }
+
+  if (!vm.$options) {
+    return ANONYMOUS_COMPONENT
+  }
+
+  const options = vm.$options
```
Comment (subzero10) [inline packages/vue/src/vue-debug.js:27]: I simplified this using Sentry's implementation as a reference point.

**Review by stympy — DISMISSED** [review]

File: packages/vue/src/vue-debug.js [hunk 2]
```diff
@@ -8,15 +8,27 @@ const classify = str => str
...
+  if (!options) {
+    return ANONYMOUS_COMPONENT
```
Comment (BethanyBerkowitz) [inline packages/vue/src/vue-debug.js:29]: Duplicate of lines 23-25 above?

**Review by BethanyBerkowitz — DISMISSED** [review]
One minor comment, but no need to re-review

File: packages/vue/src/vue-debug.js (hunk 2 above)
Comment (subzero10) [reply packages/vue/src/vue-debug.js:25]: Yes, nice catch! Thanks!

**Pull request merged**